### PR TITLE
Added user page with details and photos

### PR DESCRIPTION
**Pull request by redliz5808**

### CODE REVIEW

File: src/components/Collections/Collections.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  baseUrl = `${process.env.REACT_APP_API_BASE_URL}/users`
+
+  retrieveCollections = async (username) => {
+    try {
+      const { data } = await axios(
+        `${this.baseUrl}/${username}/collections?client_id=${process.env.REACT_APP_API_KEY}`
+      );
+      this.setState({ collections: data });
+      console.log(data)
```
Comment (yoloOnTheBattlefield) [inline src/components/Collections/Collections.js]: no console

File: src/components/Collections/Collections.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import axios from "axios";
+
+class Collections extends React.Component {
+  state = {
+    collections: null,
+  };
+
+  baseUrl = `${process.env.REACT_APP_API_BASE_URL}/users`
```
Comment (yoloOnTheBattlefield) [inline src/components/Collections/Collections.js:9]: smart

File: src/components/Photos/Photos.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  componentDidMount() {
+    this.retrieveUserPhotos(this.props.username);
+  }
+
+  render() {
+    const { photos } = this.state;
+
+    return (
+      <>
+        {photos && photos.length === 0 && <div>This user has no photos</div>}{" "}
```
Comment (yoloOnTheBattlefield) [inline src/components/Photos/Photos.js]: why the `{' '}`

File: src/pages/Photo/Photo.js [hunk 4]
```diff
@@ -15,14 +16,20 @@ import {
...
 
+  loadingBar = React.createRef();
+
   retrievePhoto = async (photoid) => {
     try {
+      this.loadingBar.current.continuousStart();
+      this.setState({ isLoading: true });
       const { data } = await axios(
-        `https://api.unsplash.com/photos/${photoid}?client_id=${process.env.REACT_APP_API_KEY}`
+        `${process.env.REACT_APP_API_BASE_URL}/photos/${photoid}?client_id=${process.env.REACT_APP_API_KEY}`
```
Comment (yoloOnTheBattlefield) [inline src/pages/Photo/Photo.js]: why not use this.baseURL again?

File: src/pages/User/User.js [hunk 5]
```diff
@@ -1,5 +1,86 @@
...
+  loadingBar = React.createRef();
+
+  retrieveUserData = async (username) => {
+    try {
+      this.loadingBar.current.continuousStart();
+      this.setState({ isLoading: true });
+      const { data } = await axios(
+        `${this.baseUrl}/${username}?client_id=${process.env.REACT_APP_API_KEY}`
+      );
+      console.log(data);
```
Comment (yoloOnTheBattlefield) [inline src/pages/User/User.js]: no console

File: src/pages/User/User.js [hunk 6]
```diff
@@ -1,5 +1,86 @@
...
+              @{data.instagram_username}
+            </InstagramUser>
+            <Paper square>
+              <Tabs
+                value={value}
+                textColor="primary"
+                indicatorColor="primary"
+                onChange={(event, newValue) => {
+                  this.setState({ value: newValue });
+                }}
```
Comment (yoloOnTheBattlefield) [inline src/pages/User/User.js]: define handle change in the body of the class. Render will create this arrow function every single time and it's bad for performance. Also we don't want to call set state from render.

https://stackoverflow.com/questions/48699573/correct-use-of-arrow-functions-in-react

File: src/pages/User/User.js [hunk 7]
```diff
@@ -1,5 +1,86 @@
...
+              >
+                <Tab label="Photos" />
+                <Tab label="Collections" />
+                <Tab label="Stats" />
+              </Tabs>
+              {value === 0 && <Photos username={this.state.username} />}
```
Comment (yoloOnTheBattlefield) [inline src/pages/User/User.js]: `const isPhotos = value === 0`

Comment (yoloOnTheBattlefield) [reply src/pages/User/User.js]: I think is slicker

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/components/Collections/Collections.js (hunk 1 above)
Comment (redliz5808) [reply src/components/Collections/Collections.js]: Seems to be a reoccurring issue with me. 😒

File: src/components/Collections/Collections.js (hunk 2 above)
Comment (redliz5808) [reply src/components/Collections/Collections.js:9]: Thanks!

File: src/pages/Photo/Photo.js (hunk 4 above)
Comment (redliz5808) [reply src/pages/Photo/Photo.js]: 🤦‍♀️

File: src/pages/User/User.js (hunk 5 above)
Comment (redliz5808) [reply src/pages/User/User.js]: 🤦‍♀️

File: src/pages/User/User.js (hunk 6 above)
Comment (redliz5808) [reply src/pages/User/User.js]: This was how they set it up in the docs, but I like your way better. Thanks!

File: src/pages/User/User.js (hunk 7 above)
Comment (redliz5808) [reply src/pages/User/User.js]: I think I did what you want...will have you double-check before I merge.

File: src/components/Photos/Photos.js (hunk 3 above)
Comment (redliz5808) [reply src/components/Photos/Photos.js]: I don't know. That somehow was added to a couple of my files and I must have missed removing this one.

**Pull request merged**